include/swift/AST/DiagnosticsModuleDiffer.def

```diff
@@ -68,6 +68,10 @@ ERROR(default_associated_type_removed,none,"%0 has removed default type %1", (St
 
 ERROR(protocol_req_added,none,"%0 has been added as a protocol requirement", (StringRef))
 
+ERROR(super_class_removed,none,"%0 has removed its super class %1", (StringRef, StringRef))
+
+ERROR(super_class_changed,none,"%0 has changed its super class from %1 to %2", (StringRef, StringRef, StringRef))
```

Comment:
This is only an API break if the new superclass is not a subclass of the old one.

Reply:
ah, I'll filter them out under ABI mode.

Reply:
I'm not sure what you mean. Imagine if v1 of a framework has the following:

```
class A {}
class C : A {}
```

And v2 has

```
class A {}
class B : A {}
class C : B {}
```

Then v2 is an API-compatible an ABI-compatible evolution of v1.

Reply:
I take it back. I think the situation is always benign if we inject a type in the inheritance chain. The current module dump needs additional information to analyze this situation. Before we implement this refinement, the framework authors can explicitly mark this super class change as a false positive.

Reply:
I think that's a good idea. Thanks!